guide/getting-started-with-testcontainers-for-dotnet/index.adoc

```diff
@@ -0,0 +1,155 @@
+bootstrapping integration tests with real services wrapped in Docker
+containers. Using Testcontainers, you can write tests that talk to the same
+type of services you use in production, without mocks or in-memory services.
+
+[NOTE]
+If you are new to Testcontainers, please read
+https://testcontainers.com/guides/introducing-testcontainers[What is
+Testcontainers, and why should you use it?] to learn more about it.
+
+Let us look at how we can use Testcontainers to test a .NET application using a
+Postgres database.
+
+== Create a solution file with a source and test project
+
+Create a .NET source and test project from your favorite IDE. Once the projects
+are created, add the following dependencies to the source and test project.
+
+.Source project
+[source,xml,indent=0]
+----
+include::{codebase}/src/GettingStarted/GettingStarted.csproj[lines="6"]
+----
+
+.Test project
+[source,xml,indent=0]
+----
+include::{codebase}/tests/GettingStarted.Tests/GettingStarted.Tests.csproj[lines="6..8"]
+----
+
+We have added *Npgsql*, a Postgres ADO.NET Data Provider for talking to the
+Postgres database, and *xUnit.net* for testing our service.
+
+== Implement business logic
+
+We are going to create a `CustomerService` class to manage customer details.
+
+First, let us create a `Customer` type as follows:
+
+[source%nowrap,csharp]
+----
+include::{codebase}/src/GettingStarted/Customer.cs[]
+----
+
+Create a `DbConnectionProvider` class to hold ADO.NET connection parameters
+(connection string) and create a method to get a database connection, as follows:
+
+[source%nowrap,csharp]
+----
+include::{codebase}/src/GettingStarted/DbConnectionProvider.cs[]
+----
+
+Create a `CustomerService` class and add the following code:
+
+[source%nowrap,csharp]
+----
+include::{codebase}/src/GettingStarted/CustomerService.cs[]
+----
+
+Let us understand what is going on in the `CustomerService` class.
+
+* We are calling `_dbConnectionProvider.GetConnection()` to get a database
+  connection using ADO.NET.
+* We have the `CreateCustomersTable()` method that creates the `customers`
+  table if it does not already exist.
+* We have the `GetCustomers()` method that fetches all rows from the
+  `customers` table, populates data into `Customer` objects, and returns a list
+  of `Customer` objects.
+* We have the `Create(Customer)` method that inserts a new customer record into
+  the database.
+
+Now let us see how we can test the `CustomerService` logic using Testcontainers.
+
+== Add Testcontainers dependencies
+
+Before writing Testcontainers-based tests, let's add Testcontainers
+dependencies to the test project as follows:
+
+.Test project
+[source,xml,indent=0]
+----
+include::{codebase}/tests/GettingStarted.Tests/GettingStarted.Tests.csproj[lines="9"]
+----
+
+As we are using a Postgres database for our application, we added the
+Testcontainers Postgres
+https://www.nuget.org/packages/Testcontainers.PostgreSql[module] as a test
```

Comment:
Nit: Add the link for "Postgres module" instead of just "module" as follows:

Testcontainers https://www.nuget.org/packages/Testcontainers.PostgreSql[Postgres module] as a test dependency.

Reply:
I've changed the link text.